File: 1209.py
import sys
from collections import deque


linha = sys.stdin.readline()
while linha:
    # Lendo valores de entrada
    n, m, k = map(int, linha.split())
    
    # Inicializando estruturas de dados
    f = [0] * (n + 1)  # Array que armazena a quantidade de conexões de cada nó
    vis = [False] * (n + 1)  # Array que marca os nós visitados
    adj = [[False] * (n + 1) for _ in range(n + 1)]  # Matriz de adjacência
    
    # Lendo as conexões
    i = 0
    while i < m:
        u, v = map(int, sys.stdin.readline().split())
        adj[u][v] = adj[v][u] = True
        f[u] += 1
        f[v] += 1
        i += 1
    
    # Fila para BFS
    queue = deque()
    
    # Adicionando nós que têm menos que k conexões
    i = 1
    while i <= n:
        if f[i] < k:
            queue.append(i)
            vis[i] = True
        i += 1
    
    # Removendo nós da fila e ajustando conexões
    while queue:
        u = queue.popleft()
        i = 1
        while i <= n:
            if adj[u][i] and not vis[i]:
                f[i] -= 1
                if f[i] < k:
                    queue.append(i)
                    vis[i] = True
            i += 1
    
    # Criando a resposta com os nós restantes
    ans = []
    i = 1
    while i <= n:
        if not vis[i]:
            ans.append(i)
        i += 1
    
    # Imprimindo a resposta
    if ans:
        print(" ".join(map(str, ans)))
    else:
        print(0)
    
    linha = sys.stdin.readline()

File: 1209.c
#include <stdio.h>
#include <stdlib.h>
#include <stdbool.h>

#define MAX_N 1000

int main() {
    int n, m, k;
    while (scanf("%d %d %d", &n, &m, &k) != EOF) {
        int f[MAX_N + 1] = {0};  // Vetor de graus
        bool vis[MAX_N + 1] = {false};  //Vetor de visitados
        bool adj[MAX_N + 1][MAX_N + 1] = {false};  //Matriz de adjacência
        
        for (int i = 0; i < m; i++) {
            int u, v;
            scanf("%d %d", &u, &v);
            adj[u][v] = adj[v][u] = true;
            f[u]++;
            f[v]++;
        }
        
        int queue[MAX_N];
        int front = 0, rear = 0;
        
        for (int i = 1; i <= n; i++) {
            if (f[i] < k) {
                queue[rear++] = i;
                vis[i] = true;
            }
        }
        
        while (front < rear) {
            int u = queue[front++];
            for (int i = 1; i <= n; i++) {
                if (adj[u][i] && !vis[i]) {
                    f[i]--;
                    if (f[i] < k) {
                        queue[rear++] = i;
                        vis[i] = true;
                    }
                }
            }
        }
        
        int ans[MAX_N];
        int ans_size = 0;
        for (int i = 1; i <= n; i++) {
            if (!vis[i]) {
                ans[ans_size++] = i;
            }
        }
        
        if (ans_size > 0) {
            for (int i = 0; i < ans_size; i++) {
                if (i > 0) printf(" ");
                printf("%d", ans[i]);
            }
            printf("\n");
        } else {
            printf("0\n");
        }
    }
    return 0;
}




